### PR TITLE
[#574] Usernames

### LINKED ISSUE #574: [FEATURE] Usernames
**Is your feature request related to a problem? Please describe.**
Currently we do not have an easy human-readable identifier associated with accounts. This can make it difficult for users to find and interact with each other.

**Describe the solution you'd like**
Add an optional `username` field in the account model that contains a user-specified username which must be unique within the app/org. This field may be set on sign-up and changed at any time. 

**Note:** No data should be associated with this identifier anywhere else in this system since it may change at any point.

### PR DESCRIPTION
**Pull request by roberlander2**
## Description
This PR adds usernames to accounts and allows them to be set on signup and admin account creation or updated at any time using the `PUT` `/account/username` APIs. This PR also updates the indexes on the `accounts` database to make queries more efficient.

**Resolves #574**

[comment]: # (This project only accepts pull requests related to open issues.)
[comment]: # (If suggesting a new feature or change, please discuss it in an issue first.)
[comment]: # (If fixing a bug, there should be an issue describing it with steps to reproduce.)

## Review Time Estimate
_Please give your idea of how soon this pull request needs to be reviewed by selecting one of the options below. This can be based on the criticality of the issue at hand and/or other relevant factors._

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] Immediately
- [ ] Within a week
- [x] When possible

## Type of changes
_Please select a relevant option:_

[comment]: # (To select an option, please put an 'x' in the applicable box.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [ ] Bug fix (non-breaking change which fixes an issue).
- [x] New feature (non-breaking change which adds functionality).
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected).
- [ ] Other (any another change that does not fall in one of the above categories.)

## Checklist:
_Please select all applicable options:_

[comment]: # (To select your options, please put an 'x' in the all boxes that apply.)
[comment]: # (If you're unsure about any of these, don't hesitate to ask. We're here to help!.)

- [x] I have signed the Rokwire Contributor License Agreement ([CLA](https://rokwire.org/rokwire_cla)). (_Any contributor who is not an employee of the University of Illinois whose official duties include contributing to the Rokwire software, or who is not paid by the Rokwire project, needs to sign the CLA before their contribution can be accepted._)
- [x] I have updated the [CHANGELOG](../CHANGELOG.md).
- [x] I have read the [Contributor Guidelines](../CONTRIBUTING.md).
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My change requires updating the documentation.
- [x] I have made necessary changes to the documentation.
- [ ] I have added tests related to my changes.
- [x] My changes generate no new warnings.
- [x] New and existing unit tests pass locally with my changes.
- [x] Any dependent changes have been merged and published in downstream modules.

---

[comment]: # (Template credit: This pull request template is based on Embedded Artistry {https://github.com/embeddedartistry/templates/blob/master/.github/PULL_REQUEST_TEMPLATE.md}, Clowder {https://github.com/clowder-framework/clowder/blob/develop/.github/PULL_REQUEST_TEMPLATE.md}, and TalAter {https://github.com/TalAter/open-source-templates} templates.)

### CODE REVIEW

File: driver/web/docs/schemas/apis/shared/requests/login/Request.yaml [hunk 1]
```diff
@@ -38,3 +38,5 @@ properties:
   preferences:
     type: object
     nullable: true
+  username:
```
Comment (shurwit) [inline driver/web/docs/schemas/apis/shared/requests/login/Request.yaml:41]: Let's make the `username` nullable wherever it is not required

**Review by shurwit — COMMENTED** [review]
Thanks @roberlander2, this is looking good! Just a few minor comments below

**Review by petyos — DISMISSED** [review]
LGTM

**Review by shurwit — APPROVED** [review]
Thanks @roberlander2, LGTM!

**Pull request merged**